rules/windows/persistence_dontexpirepasswd_account.toml

```diff
@@ -0,0 +1,54 @@
+[metadata]
+creation_date = "2022/02/22"
+maturity = "production"
+updated_date = "2022/02/22"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects the creation and modification of an account with the "Don't Expire Password" option Enabled. Attackers can abuse
+this misconfiguration to persist in the domain and maintain long-term access using compromised accounts with this
+property.
+"""
+false_positives = [
+    """
+    User accounts can be used as service accounts and have their password set never to expire. This is a bad security
+    practice that exposes the account to Credential Access attacks. For cases in which user accounts cannot be avoided,
+    Microsoft provides the Group Managed Service Accounts (gMSA) feature, which ensures that the account password is
+    robust and changed regularly and automatically.
+    """,
+]
+from = "now-9m"
+index = ["winlogbeat-*", "logs-system.*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Account configured with never Expiring Password"
+references = [
+    "https://www.cert.ssi.gouv.fr/uploads/guide-ad.html#dont_expire",
+    "https://blog.menasec.net/2019/02/threat-hunting-26-persistent-password.html",
+]
+risk_score = 47
+rule_id = "62a70f6f-3c37-43df-a556-f64fa475fba2"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Persistence", "Active Directory"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.action:"modified-user-account" and event.code:"4738" and message:"'Don't Expire Password' - Enabled" and not user.id:"S-1-5-18"
```

Comment:
The example event you provided has `"NewUacValue": "0x211"` which means `Account Disabled, Password Never Expires`. This may be a FP.  

According to [menasec](https://blog.menasec.net/2019/02/threat-hunting-26-persistent-password.html) that you referenced:

We are looking for the following combination values:

-  Old UAC Value : 0x10  -> New UAC Value: 0x210
-  Old UAC Value : 0x11 -> New UAC Value: 0x210
-  Old UAC Value : 0x15 -> New UAC Value: 0x210

So we can probably use the `NewUacValue` and `OldUacValue`

Reply:
In reality UAC values can be so much more complex than these combinations, you would need to account for all possible combinations of the uac values listed here: https://docs.microsoft.com/en-us/troubleshoot/windows-server/identity/useraccountcontrol-manipulate-account-properties#list-of-property-flags

I saw inconsistences, like in this case, where the option selected didn't reflected the value on the NewUacValue nor NewUACList fields

Reply:
Maybe you can try to reproduce this change in your env and we can get more context

Reply:
Looks like i'm getting similar data on this test machine. It shows up as `0x211`, and the account is definitely enabled. 

<details>
<summary>Additional Event </summary>

```

{
  "_index": ".ds-logs-system.security-default-2022.02.16-000001",
  "_type": "_doc",
  "_id": "CDZuZX8BWnJcw9OZTuwW",
  "_version": 1,
  "_score": 1,
  "_source": {
    "agent": {
      "hostname": "dc01",
      "name": "dc01",
      "id": "d8a9e881-4cf9-4807-ade3-8f8f1794bdcc",
      "ephemeral_id": "87b69e29-de20-4ccc-806a-1bd2a05b90f9",
      "type": "filebeat",
      "version": "7.16.2"
    },
    "winlog": {
      "computer_name": "dc01.arabica.labs",
      "process": {
        "pid": 580,
        "thread": {
          "id": 1608
        }
      },
      "keywords": [
        "Audit Success"
      ],
      "logon": {
        "id": "0x55a1eae4"
      },
      "channel": "Security",
      "event_data": {
        "TargetSid": "S-1-5-21-1924272405-1406338840-2483987062-1125",
        "NewUACList": [
          "LOCKOUT",
          "SCRIPT",
          "NORMAL_ACCOUNT"
        ],
        "SidHistory": "-",
        "LogonHours": "-",
        "ScriptPath": "-",
        "DisplayName": "-",
        "HomePath": "-",
        "SubjectDomainName": "ARABICA",
        "AllowedToDelegateTo": "-",
        "TargetDomainName": "ARABICA",
        "PrivilegeList": "-",
        "UserWorkstations": "-",
        "SamAccountName": "-",
        "OldUacValue": "0x15",
        "HomeDirectory": "-",
        "UserParameters": "-",
        "NewUacValue": "0x211",
        "SubjectLogonId": "0x55a1eae4",
        "PrimaryGroupId": "-",
        "AccountExpires": "-",
        "ProfilePath": "-",
        "UserAccountControl": [
          "2050",
          "2089"
        ],
        "SubjectUserName": "test_user",
        "PasswordLastSet": "-",
        "UserPrincipalName": "-",
        "TargetUserName": "delete",
        "Dummy": "-",
        "SubjectUserSid": "S-1-5-21-1924272405-1406338840-2483987062-1124"
      },
      "opcode": "Info",
      "record_id": "279444",
      "task": "User Account Management",
      "event_id": "4738",
      "provider_guid": "{54849625-5478-4994-A5BA-3E3B0328C30D}",
      "api": "wineventlog",
      "provider_name": "Microsoft-Windows-Security-Auditing"
    },
    "log": {
      "level": "information"
    },
    "elastic_agent": {
      "id": "d8a9e881-4cf9-4807-ade3-8f8f1794bdcc",
      "version": "7.16.2",
      "snapshot": false
    },
    "message": "A user account was changed.\n\nSubject:\n\tSecurity ID:\t\tS-1-5-21-1924272405-1406338840-2483987062-1124\n\tAccount Name:\t\ttest_user\n\tAccount Domain:\t\tARABICA\n\tLogon ID:\t\t0x55A1EAE4\n\nTarget Account:\n\tSecurity ID:\t\tS-1-5-21-1924272405-1406338840-2483987062-1125\n\tAccount Name:\t\tdelete\n\tAccount Domain:\t\tARABICA\n\nChanged Attributes:\n\tSAM Account Name:\t-\n\tDisplay Name:\t\t-\n\tUser Principal Name:\t-\n\tHome Directory:\t\t-\n\tHome Drive:\t\t-\n\tScript Path:\t\t-\n\tProfile Path:\t\t-\n\tUser Workstations:\t-\n\tPassword Last Set:\t-\n\tAccount Expires:\t\t-\n\tPrimary Group ID:\t-\n\tAllowedToDelegateTo:\t-\n\tOld UAC Value:\t\t0x15\n\tNew UAC Value:\t\t0x211\n\tUser Account Control:\t\n\t\t'Password Not Required' - Disabled\n\t\t'Don't Expire Password' - Enabled\n\tUser Parameters:\t-\n\tSID History:\t\t-\n\tLogon Hours:\t\t-\n\nAdditional Information:\n\tPrivileges:\t\t-",
    "cloud": {
      "availability_zone": "us-east4-a",
      "instance": {
        "name": "stryker-dc01",
        "id": "8942387728962675673"
      },
      "provider": "gcp",
      "machine": {
        "type": "e2-standard-2"
      },
      "service": {
        "name": "GCE"
      },
      "project": {
        "id": "elastic-security-dev"
      },
      "account": {
        "id": "elastic-security-dev"
      }
    },
    "input": {
      "type": "winlog"
    },
    "@timestamp": "2022-03-07T17:29:59.694Z",
    "ecs": {
      "version": "1.12.0"
    },
    "related": {
      "user": [
        "test_user",
        "delete"
      ]
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "system.security"
    },
    "host": {
      "hostname": "dc01",
      "os": {
        "build": "14393.4946",
        "kernel": "10.0.14393.4946 (rs1_release.220131-0721)",
        "name": "Windows Server 2016 Datacenter",
        "type": "windows",
        "family": "windows",
        "version": "10.0",
        "platform": "windows"
      },
      "ip": [
        "192.168.90.5"
      ],
      "name": "dc01.arabica.labs",
      "id": "2618db7a-2692-4e04-94e0-839fddf5578b",
      "mac": [
        "42:01:c0:a8:5a:05"
      ],
      "architecture": "x86_64"
    },
    "event": {
      "agent_id_status": "verified",
      "ingested": "2022-03-07T17:30:02Z",
      "code": "4738",
      "provider": "Microsoft-Windows-Security-Auditing",
      "created": "2022-03-07T17:30:01.119Z",
      "kind": "event",
      "action": "modified-user-account",
      "category": [
        "iam"
      ],
      "type": [
        "user",
        "change"
      ],
      "dataset": "system.security",
      "outcome": "success"
    },
    "user": {
      "domain": "ARABICA",
      "name": "test_user",
      "id": "S-1-5-21-1924272405-1406338840-2483987062-1124",
      "target": {
        "domain": "ARABICA",
        "name": "delete",
        "id": "S-1-5-21-1924272405-1406338840-2483987062-1125"
      }
    }
  },
  "fields": {
    "winlog.event_data.SidHistory": [
      "-"
    ],
    "elastic_agent.version": [
      "7.16.2"
    ],
    "event.category": [
      "iam"
    ],
    "winlog.event_data.NewUACList": [
      "LOCKOUT",
      "SCRIPT",
      "NORMAL_ACCOUNT"
    ],
    "host.os.name.text": [
      "Windows Server 2016 Datacenter"
    ],
    "host.hostname": [
      "dc01"
    ],
    "host.mac": [
      "42:01:c0:a8:5a:05"
    ],
    "winlog.process.pid": [
      580
    ],
    "winlog.event_data.UserParameters": [
      "-"
    ],
    "host.os.version": [
      "10.0"
    ],
    "winlog.logon.id": [
      "0x55a1eae4"
    ],
    "agent.name": [
      "dc01"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "event.outcome": [
      "success"
    ],
    "winlog.event_data.UserWorkstations": [
      "-"
    ],
    "user.id": [
      "S-1-5-21-1924272405-1406338840-2483987062-1124"
    ],
    "host.os.type": [
      "windows"
    ],
    "input.type": [
      "winlog"
    ],
    "agent.hostname": [
      "dc01"
    ],
    "user.target.name": [
      "delete"
    ],
    "related.user": [
      "test_user",
      "delete"
    ],
    "host.architecture": [
      "x86_64"
    ],
    "event.provider": [
      "Microsoft-Windows-Security-Auditing"
    ],
    "cloud.provider": [
      "gcp"
    ],
    "cloud.machine.type": [
      "e2-standard-2"
    ],
    "event.code": [
      "4738"
    ],
    "agent.id": [
      "d8a9e881-4cf9-4807-ade3-8f8f1794bdcc"
    ],
    "winlog.event_data.SubjectUserSid": [
      "S-1-5-21-1924272405-1406338840-2483987062-1124"
    ],
    "winlog.process.thread.id": [
      1608
    ],
    "winlog.event_data.PrivilegeList": [
      "-"
    ],
    "user.name": [
      "test_user"
    ],
    "winlog.event_data.AccountExpires": [
      "-"
    ],
    "cloud.instance.id": [
      "8942387728962675673"
    ],
    "host.ip": [
      "192.168.90.5"
    ],
    "agent.type": [
      "filebeat"
    ],
    "winlog.event_data.TargetSid": [
      "S-1-5-21-1924272405-1406338840-2483987062-1125"
    ],
    "winlog.event_data.SubjectLogonId": [
      "0x55a1eae4"
    ],
    "winlog.event_data.ProfilePath": [
      "-"
    ],
    "winlog.api": [
      "wineventlog"
    ],
    "winlog.event_data.NewUacValue": [
      "0x211"
    ],
    "user.target.domain": [
      "ARABICA"
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "user.domain": [
      "ARABICA"
    ],
    "host.id": [
      "2618db7a-2692-4e04-94e0-839fddf5578b"
    ],
    "elastic_agent.id": [
      "d8a9e881-4cf9-4807-ade3-8f8f1794bdcc"
    ],
    "winlog.event_data.LogonHours": [
      "-"
    ],
    "event.ingested": [
      "2022-03-07T17:30:02.000Z"
    ],
    "event.action": [
      "modified-user-account"
    ],
    "@timestamp": [
      "2022-03-07T17:29:59.694Z"
    ],
    "winlog.channel": [
      "Security"
    ],
    "host.os.platform": [
      "windows"
    ],
    "cloud.account.id": [
      "elastic-security-dev"
    ],
    "winlog.event_data.TargetDomainName": [
      "ARABICA"
    ],
    "data_stream.dataset": [
      "system.security"
    ],
    "winlog.opcode": [
      "Info"
    ],
    "agent.ephemeral_id": [
      "87b69e29-de20-4ccc-806a-1bd2a05b90f9"
    ],
    "winlog.event_data.SubjectDomainName": [
      "ARABICA"
    ],
    "winlog.event_data.Dummy": [
      "-"
    ],
    "cloud.instance.name": [
      "stryker-dc01"
    ],
    "cloud.project.id": [
      "elastic-security-dev"
    ],
    "winlog.event_data.UserAccountControl": [
      "2050",
      "2089"
    ],
    "winlog.provider_guid": [
      "{54849625-5478-4994-A5BA-3E3B0328C30D}"
    ],
    "winlog.provider_name": [
      "Microsoft-Windows-Security-Auditing"
    ],
    "winlog.computer_name": [
      "dc01.arabica.labs"
    ],
    "user.target.id": [
      "S-1-5-21-1924272405-1406338840-2483987062-1125"
    ],
    "cloud.availability_zone": [
      "us-east4-a"
    ],
    "winlog.keywords": [
      "Audit Success"
    ],
    "winlog.record_id": [
      "279444"
    ],
    "host.os.name": [
      "Windows Server 2016 Datacenter"
    ],
    "log.level": [
      "information"
    ],
    "winlog.event_data.AllowedToDelegateTo": [
      "-"
    ],
    "winlog.event_data.HomePath": [
      "-"
    ],
    "host.name": [
      "dc01.arabica.labs"
    ],
    "event.kind": [
      "event"
    ],
    "winlog.event_data.OldUacValue": [
      "0x15"
    ],
    "winlog.event_data.TargetUserName": [
      "delete"
    ],
    "data_stream.type": [
      "logs"
    ],
    "cloud.service.name": [
      "GCE"
    ],
    "ecs.version": [
      "1.12.0"
    ],
    "event.created": [
      "2022-03-07T17:30:01.119Z"
    ],
    "winlog.event_data.ScriptPath": [
      "-"
    ],
    "agent.version": [
      "7.16.2"
    ],
    "host.os.family": [
      "windows"
    ],
    "winlog.event_data.SamAccountName": [
      "-"
    ],
    "winlog.event_data.PrimaryGroupId": [
      "-"
    ],
    "winlog.event_data.DisplayName": [
      "-"
    ],
    "host.os.build": [
      "14393.4946"
    ],
    "event.module": [
      "system"
    ],
    "host.os.kernel": [
      "10.0.14393.4946 (rs1_release.220131-0721)"
    ],
    "winlog.event_data.UserPrincipalName": [
      "-"
    ],
    "winlog.task": [
      "User Account Management"
    ],
    "winlog.event_data.HomeDirectory": [
      "-"
    ],
    "winlog.event_data.SubjectUserName": [
      "test_user"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "message": [
      "A user account was changed.\n\nSubject:\n\tSecurity ID:\t\tS-1-5-21-1924272405-1406338840-2483987062-1124\n\tAccount Name:\t\ttest_user\n\tAccount Domain:\t\tARABICA\n\tLogon ID:\t\t0x55A1EAE4\n\nTarget Account:\n\tSecurity ID:\t\tS-1-5-21-1924272405-1406338840-2483987062-1125\n\tAccount Name:\t\tdelete\n\tAccount Domain:\t\tARABICA\n\nChanged Attributes:\n\tSAM Account Name:\t-\n\tDisplay Name:\t\t-\n\tUser Principal Name:\t-\n\tHome Directory:\t\t-\n\tHome Drive:\t\t-\n\tScript Path:\t\t-\n\tProfile Path:\t\t-\n\tUser Workstations:\t-\n\tPassword Last Set:\t-\n\tAccount Expires:\t\t-\n\tPrimary Group ID:\t-\n\tAllowedToDelegateTo:\t-\n\tOld UAC Value:\t\t0x15\n\tNew UAC Value:\t\t0x211\n\tUser Account Control:\t\n\t\t'Password Not Required' - Disabled\n\t\t'Don't Expire Password' - Enabled\n\tUser Parameters:\t-\n\tSID History:\t\t-\n\tLogon Hours:\t\t-\n\nAdditional Information:\n\tPrivileges:\t\t-"
    ],
    "winlog.event_id": [
      "4738"
    ],
    "event.type": [
      "user",
      "change"
    ],
    "winlog.event_data.PasswordLastSet": [
      "-"
    ],
    "event.dataset": [
      "system.security"
    ]
  }
}

```

</details>